oxd-server/src/main/java/org/gluu/oxd/server/op/GetOpDiscoveryConfigOperation.java

```diff
@@ -0,0 +1,47 @@
+package org.gluu.oxd.server.op;
+
+import org.apache.commons.beanutils.BeanUtils;
+import org.gluu.oxauth.client.OpenIdConfigurationResponse;
+import org.gluu.oxd.common.ErrorResponseCode;
+import org.gluu.oxd.common.response.GetOpDiscoveryConfigResponse;
+import org.gluu.oxd.server.HttpException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.inject.Injector;
+import org.gluu.oxd.common.Command;
+import org.gluu.oxd.common.params.GetOpDiscoveryConfigParams;
+import org.gluu.oxd.common.response.IOpResponse;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+
+/**
+ * @author Yuriy Zabrovarnyy
+ * @version 0.9, 22/09/2015
+ */
+
+public class GetOpDiscoveryConfigOperation extends BaseOperation<GetOpDiscoveryConfigParams> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GetOpDiscoveryConfigOperation.class);
+
+    /**
+     * Base constructor
+     *
+     * @param command command
+     */
+    protected GetOpDiscoveryConfigOperation(Command command, final Injector injector) {
+        super(command, injector, GetOpDiscoveryConfigParams.class);
+    }
+
+    public IOpResponse execute(GetOpDiscoveryConfigParams params) throws IOException {
+        OpenIdConfigurationResponse discoveryResponse = getDiscoveryService().getConnectDiscoveryResponse(params.getOpHost(), params.getOpDiscoveryPath());
+
```

Comment:
Can we remove `OpenIdConfiguraionResponse` and use `org.gluu.oxauth.model.discovery.OAuth2Discovery` instead ?

Reply:
1. I found in OAuth2Discovery only below fields are present. There are many fields which are not present. Please confirm if this is ok.
        "issuer",
        "authorization_endpoint",
        "token_endpoint",
        "jwks_uri",
        "registration_endpoint",
        "response_types_supported",
        "response_modes_supported",
        "grant_types_supported",
        "token_endpoint_auth_methods_supported",
        "token_endpoint_auth_signing_alg_values_supported",
        "service_documentation",
        "ui_locales_supported",
        "op_policy_uri",
        "op_tos_uri",
        "revocation_endpoint",
        "revocation_endpoint_auth_methods_supported",
        "revocation_endpoint_auth_signing_alg_values_supported",
        "introspection_endpoint",
        "introspection_endpoint_auth_methods_supported",
        "introspection_endpoint_auth_signing_alg_values_supported",
        "code_challenge_methods_supported"

2.  Do you want to create a new method in org.gluu.oxd.server.service.DiscoveryService.java returning org.gluu.oxauth.model.discovery.OAuth2Discovery? Please let me know if you have any different idea.

Reply:
We have `OpenIdConfigurationResponse` on oxd which I believe is enough. I think oxauth need to expose common class which can be re-used. Right now we are forced to introduce one more class. So lets leave it as is. I've opened improvement ticket to enhance it on oxauth first and then we will drop `GetOpDiscoveryConfigResponse ` from oxd.  Ticket: https://github.com/GluuFederation/oxAuth/issues/1079